tests/test_timing_properties.py

```diff
@@ -0,0 +1,143 @@
+        with nidaqmx.Task() as task:
+            task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+
+            task.timing.cfg_samp_clk_timing(1000)
+
+            default_value = task.timing.samp_clk_src
+
+            # Test property default value.
+            assert task.timing.samp_clk_src == default_value
+
+            # Test property setter and getter.
+            task.timing.samp_clk_src = "PFI0"
+            assert task.timing.samp_clk_src == "/nidaqmxMultithreadingTester1/PFI0"
+
+            # Test property deleter.
+            del task.timing.samp_clk_src
+            assert task.timing.samp_clk_src == default_value
+
+            # Test Invalid terminal. Reading this property throws error.
+            task.timing.cfg_samp_clk_timing(1000, source="Test_Invalid_Device_Source")
+            with pytest.raises(DaqError) as e:
+                _ = task.timing.samp_clk_src
+            assert e.value.error_code == -89120
+
+    def test_enum_property(self, any_x_series_device):
+        """Test for validating enum property."""
+        with nidaqmx.Task() as task:
+            task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+
+            task.timing.cfg_samp_clk_timing(1000, sample_mode=AcquisitionType.CONTINUOUS)
+
+            # Test property initial value.
+            assert task.timing.samp_quant_samp_mode == AcquisitionType.CONTINUOUS
+
+            # Test property setter and getter.
+            task.timing.samp_quant_samp_mode = AcquisitionType.FINITE
+            assert task.timing.samp_quant_samp_mode == AcquisitionType.FINITE
+
+            # Test property deleter.
+            del task.timing.samp_quant_samp_mode
+            assert task.timing.samp_quant_samp_mode == AcquisitionType.CONTINUOUS
+
+    def test_float64_property(self, any_x_series_device):
+        """Test for validating float64 property."""
+        with nidaqmx.Task() as task:
+            task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+
+            task.timing.cfg_samp_clk_timing(1000)
+
+            # Test property initial value.
+            assert task.timing.samp_clk_rate == 1000
+
+            # Test property setter and getter.
+            task.timing.samp_clk_rate = 2000
+            assert task.timing.samp_clk_rate == 2000
+
+            # Test property deleter.
+            del task.timing.samp_clk_rate
+            assert task.timing.samp_clk_rate == 1000
+
+    def test_uint32_property(self, any_x_series_device):
+        """Test for validating unsigned int32 property."""
+        with nidaqmx.Task() as task:
+            task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+
+            task.timing.cfg_samp_clk_timing(1000)
+
+            # Test property initial value.
+            default_value = task.timing.samp_clk_timebase_div
+            assert task.timing.samp_clk_timebase_div == default_value
```

Comment:
this doesn't seem useful since you just got the default

Reply:
Hardcoded the expected actual values.